File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.js
let cnt = 0;
function solution(numbers, target) {
    dfs(numbers, target, 0, 0);
    return cnt;
}

function dfs(numbers, target, idx, number){
    if (idx===numbers.length){
        if (number===target) {
            cnt++;
        }
        return;
    }
    dfs(numbers, target, idx+1, number+numbers[idx]);
    dfs(numbers, target, idx+1, number-numbers[idx]);
}

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.java
class Solution {
    static int cnt = 0;
    public int solution(int[] numbers, int target) {
        int answer = 0;
        
        dfs(numbers, target, 0, 0);
        
        return cnt;
    }
    static void dfs(int[] numbers, int target, int idx, int result) {
        if (idx==numbers.length) {
            if (result==target) {
                cnt++;
            }
            return;
        }
        
        dfs(numbers, target, idx+1, result+numbers[idx]);
        dfs(numbers, target, idx+1, result-numbers[idx]);
    }
}